services-core/src/main/java/com/mapbox/core/utils/TextUtils.java

```diff
@@ -111,27 +126,44 @@ public static String formatRadiuses(double[] radiuses) {
     return join(";", radiusesFormatted);
   }
 
+  public static String formatRadiuses(List<Double> radiuses) {
+    if (radiuses == null || radiuses.size() == 0) {
+      return null;
+    }
+
+    String[] radiusesFormatted = new String[radiuses.size()];
+    for (int i = 0; i < radiuses.size(); i++) {
+      if (radiuses.get(i) == Double.POSITIVE_INFINITY) {
+        radiusesFormatted[i] = "unlimited";
+      } else {
+        radiusesFormatted[i] = String.format(Locale.US, "%s",
+                TextUtils.formatCoordinate(radiuses.get(i)));
+      }
+    }
+    return join(";", radiusesFormatted);
+  }
+
   /**
    * Formats the bearing variables from the raw values to a string which can than be used for the
    * request URL.
    *
-   * @param bearings a List of doubles representing bearing values
+   * @param bearings a List of list of doubles representing bearing values
    * @return a string with the bearing values
    * @since 3.0.0
    */
-  public static String formatBearing(List<Double[]> bearings) {
+  public static String formatBearing(List<List<Double>> bearings) {
```

Comment:
I can't have 2 methods
`formatBearing(List<Double[]> bearings)`
and
`formatBearing(List<List<Double>> bearings)`
because I get the next compile error:

<img width="742" alt="Screen Shot 2020-01-31 at 12 32 14" src="https://user-images.githubusercontent.com/14282440/73528332-bfc41980-4425-11ea-923c-37f6274e8809.png">

Reply:
so we can add new static parse methods to another file. To `MapboxDirections` for example
wdyt?

Reply:
How about naming the method `formatBearings`? That would make sense from the data standpoint as well.

Reply:
makes sense!